ev-dev-tools/src/ev_cli/templates/ld-ev.cpp.j2

```diff
@@ -93,12 +93,21 @@ std::vector<Everest::cmd> everest_register(const json& connections) {
     {% if requirement.is_vector %}
     auto r_{{ requirement.id }} = std::vector<std::unique_ptr<{{ requirement.class_name }}>>();
     if (connections.contains("{{ requirement.id }}")) {
-        for (int idx = 0; idx < connections["{{ requirement.id }}"].size(); idx++) {
-            r_{{ requirement.id }}.emplace_back(std::make_unique<{{ requirement.class_name }}>(&adapter, Requirement{"{{ requirement.id }}", idx}));
+        auto requirement_connection = connections.at("{{ requirement.id }}");
+        for (size_t idx = 0; idx < requirement_connection.size(); idx++) {
+            auto requirement_module_id = requirement_connection[idx].at("module_id");
+            r_{{ requirement.id }}.emplace_back(std::make_unique<{{ requirement.class_name }}>(&adapter, Requirement{"{{ requirement.id }}", idx}, requirement_module_id));
         }
     }
     {% else %}
-    auto r_{{ requirement.id }} = std::make_unique<{{ requirement.class_name }}>(&adapter, Requirement{"{{ requirement.id }}", 0});
+    std::string r_{{ requirement.id }}_requirement_module_id;
```

Comment:
This looks unnecessary but I see the reason here. 
If requirement is not a vector (the naming is also bad here, should be called fullfillment), than we implicetely assume that there has to be at least one inside the connections.

One way to go could be:
    auto r_{{ requirement.id }} = {% if requirement.is_vector -%}
        std::vector<std::unique_ptr<{{ requirement.class_name }}>>();
    {%- else -%}
        std::unique_ptr<{{ requirement.class_name }}>();
    {%- endif %}
    {
        auto requirement_connection = connections.at("{{ requirement.id }}");
        {% if !requirement.is_vector %}
        assert(requirement_connection.size() == 1);
        {% endif %}
        for (size_t idx = 0; idx < requirement_connection.size(); idx++) {
            auto requirement_module_id = requirement_connection[idx].at("module_id");
            r_{{ requirement.id }}{$ '.emplace_back' if requirement.is_vector else ' = ' }(std::make_unique<{{ requirement.class_name }}>(&adapter, Requirement{"{{ requirement.id }}", idx}, requirement_module_id));
        }
    }

Reply:
I agree that this would create slightly less verbose auto generated code. However the template code, at least in my opinion, becomes much harder to understand with those inline ifs.
Maybe we should refactor this in another PR to separate the new feature from the refactoring?

Reply:
:+1:
probably we can add a comment like // FIXME: needs refactoring ..., so we don't forget about it